src/speakers/extraction.py

```diff
@@ -329,3 +332,312 @@ def get_speakers_in_bold(doc):
+        if len(words)<=5 or (words[0]=="Moderator" or words[0]=='Moderator:'): 
+            if len(words)==1 and len(words[0])<4:
+                flag=1
+            for word in words:
+                if not word[0].isupper():
+                    flag=1
+                for letter in word:
+                    if letter.isnumeric() or letter in char_not_in_names:
+                        flag=1
+            if flag==0:
+                if line not in names:
+                    names.append(line)
+    names=check_dict(names)
+    names=check_last_char(names)
+    names=check_repetations(names)
+    for line in lines:
+        if line in names:
+            converse.append(line)
+    return [names,lines,converse]
+
+
+def get_conversation(doc):
+    names,lines,converse=get_speaker_names(doc)
+    s=''
+    conversation=[]
+    for line in lines:
+        if line not in names:
+            s=s+line+' '
+        else:
+            conversation.append(s)
+            s=''
+    if s!='':
+        conversation.append(s)
+    conversation_from_beginning=conversation
+    conversation=conversation[1:len(conversation)]
+    return (conversation,conversation_from_beginning)
+
+
+def modify_str(l):
+    str=l.replace('.','. ')
+    l=l.replace(',',', ')
+    l=l.replace('from','from ')
+    l=l.replace('(','')
+    l=l.replace(')','.')
+    w=l.split()
+    return w
+
+
+def find_count(name,w,count_max):
+    flag=0
+    position=0
+    count=0
+    name_index=0
+    for word in name.split():
+        word=remove_last_char_if_colon(word)
+        if word in w:
+            if flag==0:
+                name_index=w.index(word)
+                flag=1
+            count=count+1
+        elif flag==0:
+            position=position+1
+    if count>count_max:
+        count_max=count
+    return (count,count_max,position,name_index)
+
+
+def check_if_firm(firm):
+    count=0
+    if len(firm.split())>4:
+        for word in firm.split():
+            if dictionary.check(word):
+                count=count+1
+        if count==len(firm.split()):
+            firm=''
+    if firm=='Thank you':
+        firm=''
+    return firm
+
+def pass_1(names,conversation,converse):
+    final={}
+    for name in names:
+        name=remove_last_char_if_colon(name)
+        count_max=0
+        for i in range(len(conversation)):
+            if converse[i]=='Moderator:' or converse[i]=='Moderator' or converse[i]=='Operator' or converse[i]=='Operator:':
+                count=0
+                l=conversation[i]
+                #modify_str takes the conversation, makes some changes and returns the words to w
+                w=modify_str(l)
+                #postion is the postion of the word identified in name 
+                #name_index is the postion of the word identified in w                
+                count,count_max,position,name_index=find_count(name,w,count_max)
+                if count/len(name.split())>=0.5 and count>=count_max:
```

Comment:
Trying to find the paragraph where the full name appears? Please add comments for non trivial checks.